src/sciline/_provider.py

```diff
@@ -303,6 +304,10 @@ def _get_func_args_and_types(
     func = _unwrap_decorated(func)
     hints = get_type_hints(func)
     signature = inspect.getfullargspec(func)
+    if sys.version_info[1] < 11:
+        for name, param in inspect.signature(func).parameters.items():
+            if param.default is None:
+                hints[name] = get_args(hints[name])[0]
```

Comment:
What happens if the function signature is `f(a: int | None = None)`? We don't want to remove the optional in that case!

Reply:
Aha, I thought an additional optional was added as a wrapper also in that case, but I guess it isn't.

Reply:
I don't know if there's a good way to fix this. Maybe better in that case to just do nothing and wait for later python versions like was suggested in the issue.

Reply:
Agree, do nothing!